Modules/_ssl.c

```diff
@@ -3167,10 +3171,14 @@ _ssl__SSLContext_impl(PyTypeObject *type, int proto_version)
        usage for no cost at all. */
     SSL_CTX_set_mode(self->ctx, SSL_MODE_RELEASE_BUFFERS);
 
+    /* Setting the session id context is a server-side only operation.
```

Comment:
Can you make a separate PR for this little logic change about only calling set_session_id_context for servers?  It seems like a bugfix that can stand on its own.

Reply:
Issue: https://github.com/python/cpython/issues/105293
PR: https://github.com/python/cpython/pull/105295

Reply:
I'm going to leave this comment unresolved as a reminder that the other PR needs to be merged first.